docs/src/main/sphinx/connector/bigquery.rst

```diff
@@ -45,24 +45,22 @@ that should generally lead to better read performance:
 Requirements
 ------------
 
-Enable the BigQuery Storage API
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Follow `these instructions <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.
-
-Authentication
-^^^^^^^^^^^^^^
-
-**On GCE/Dataproc** the authentication is taken from the machine's role.
-
-**Outside GCE/Dataproc** you have 3 options:
-
-* Use a service account JSON key and ``GOOGLE_APPLICATION_CREDENTIALS`` as
-  described `here <https://cloud.google.com/docs/authentication/getting-started>`_.
-* Set ``bigquery.credentials-key`` in the catalog properties file.
-  It should contain the contents of the JSON file, encoded using base64.
-* Set ``bigquery.credentials-file`` in the catalog properties file.
-  It should point to the location of the JSON file.
+* Enable the `BigQuery Storage Read API
+  <https://cloud.google.com/bigquery/docs/reference/storage/#enabling_the_api>`_.
```

Comment:
After this add the network access item from coordinator and workers. Not sure about any default port..

Reply:
Since this connector uses the Google Cloud API, it should just be hitting their Cloud project's service endpoint (which, as best I can tell, uses HTTPS without any custom port by default).

Reply:
This reminds me - do we document anything regarding access to HDFS/S3 for the Hive connector? Or are we only mentioning the metastore there?

Reply:
We also mention access to the storage .. its critically important for things to work..

Reply:
And in this case we need to mention that access to the Google Cloud API from the coordinator and workers is necessary.

Reply:
In that case please do mention that permissions are needed in BigQuery.
See https://cloud.google.com/bigquery/docs/reference/storage

> requires permissions to two distinct resources within BigQuery: the project that controls the session and the table from which the data is read.

Rest of the communication happens over HTTPS probably but the Google Cloud SDK doesn't call it out so I cannot be sure about that part.

Reply:
We still need to at least mention it since you could run Trino in a private network without global internet access and then the Google Cloud API has to be added to an allowlist

Reply:
Rephrased slightly to be more clear about network access to the API endpoint (through which, I believe, we get access to the storage - no separate storage network access). Also added a requirement for setting BigQuery permissions.

How's that look?